### PR TITLE
sql/analyzer: resolve ascend and descend indexes

### LINKED ISSUE #211: PilosaDriver: Implement AscendIndex and DescendIndex interfaces
First of all, we need to modify a bit the existing interface to apply all the changes that we did on the base Index interface. 
- Instead of accepting `key interface{}` we need to modify methods to use `key ...interface{}`. 
- `AscendRange(greaterOrEqual, lessThan interface{}) (IndexLookup, error)` becomes `AscendRange(greaterOrEqual, lessThan []interface{}) (IndexLookup, error)`.

The same changes should be applied to DescendIndex interface.

### LINKED ISSUE #214: analyzer: Add to resolve index rule range checking using DescendIndex and AscendIndex.
Right now we are checking for equality expressions, we need to implement the same logic but to be able to use range expressions described in DescendIndex and AscendIndex interfaces.

### PR DESCRIPTION
**Pull request by erizocosmico**
Closes #214

### CODE REVIEW

File: engine_test.go [hunk 1]
```diff
@@ -706,48 +706,93 @@ func TestStarPanic197(t *testing.T) {
 }
 
 func TestIndexes(t *testing.T) {
-	require := require.New(t)
```
Comment (ajnavarro) [inline engine_test.go]: why did you remove this?

File: engine_test.go [hunk 2]
```diff
@@ -706,48 +706,93 @@ func TestStarPanic197(t *testing.T) {
...
 	defer func() {
 		done, err := e.Catalog.DeleteIndex("foo", "myidx", true)
-		require.NoError(err)
+		require.NoError(t, err)
 		<-done
 	}()
 
-	_, it, err := e.Query(sql.NewEmptyContext(), "SELECT * FROM mytable WHERE i = 2")
-	require.NoError(err)
+	testCases := []struct {
```
Comment (ajnavarro) [inline engine_test.go]: shall we test indexes made from more than one column too?

Comment (ajnavarro) [reply engine_test.go]: and some AND and OR

File: engine_test.go (hunk 1 above)
Comment (erizocosmico) [reply engine_test.go]: Because now it's created inside each of the subtests. Otherwise, `require` is redefined and you can't do `require.New` inside

File: engine_test.go (hunk 2 above)
Comment (erizocosmico) [reply engine_test.go]: We cannot test `AND` and `OR` because there is no `SetOperations` implemented yet in pilosa

Comment (erizocosmico) [reply engine_test.go]: Added the tests for more than one column

**Comment by ajnavarro** [comment]
@erizocosmico rebase and merge please!
